rrrspec-client/lib/rrrspec/client/rspec_runner_factory.rb

```diff
@@ -0,0 +1,24 @@
+require 'rrrspec/client/rspec_runner'
+
+module RRRSpec
+  module Client
+    class RSpecRunnerFactory
+      attr_reader :next_factory
+
+      def initialize(successor)
+        @next_factory = successor
+      end
+
+      def create(file_extension)
+        unless Thread.current[:rspec3_runner]
+          Thread.current[:rspec3_runner] = RSpecRunner.new
```

Comment:
Each slave runs as a thread, this was done to ensure that there weren't different threads using the same rspec runner, I had problems with the cucumber runner because of this (the count of passed, failed and pending would get mixed). 

The rspec runner didn't seemed thread safe, but maybe the underlying rspec is it?

Slaves ares
```
Thread.fork { slave_runner.work_loop }
```